packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/invoke-model.ts

```diff
@@ -162,18 +189,25 @@ export class BedrockInvokeModel extends sfn.TaskStateBase {
       throw new Error('Output S3 object version is not supported.');
     }
 
-    this.taskPolicies = this.renderPolicyStatements();
+    //Warning to let users know about the newly introduced props
+    if (props.inputPath || props.outputPath && !isFeatureFlagEnabled) {
+      Annotations.of(scope).addWarningV2('aws-cdk-lib/aws-stepfunctions-taks',
+        'These props will set the value of inputPath/outputPath as s3 URI under input/output field in state machine JSON definition. To modify the behaviour set feature flag `@aws-cdk/aws-stepfunctions-tasks:useNewS3UriParametersForBedrockInvokeModelTask": true` and use props s3InputUri/s3OutputUri');
+    }
+
+    this.taskPolicies = this.renderPolicyStatements(isFeatureFlagEnabled);
   }
 
-  private renderPolicyStatements(): iam.PolicyStatement[] {
+  private renderPolicyStatements(isFeatureFlagEnabled?: boolean): iam.PolicyStatement[] {
```

Comment:
optional, I prefer to recalculate the `isFeatureFlagEnabled` here in this method instead of passing it, so juts in case in the future if we have another problem and we need to add a new feature flag, so we do not confuse different flags.